### PR TITLE
KYLIN-4625 Debug the code of Kylin on Parquet without hadoop environment

### PR DESCRIPTION
**Pull request by RupengWang**
## Proposed changes

Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Types of changes

What types of changes does your code introduce to Kylin?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have create an issue on [Kylin's jira](https://issues.apache.org/jira/browse/KYLIN), and have described the bug/feature there in detail
- [x] Commit messages in my PR start with the related jira ID, like "KYLIN-0000 Make Kylin project open-source"
- [x] Compiling and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If this change need a document change, I will prepare another pr against the `document` branch
- [ ] Any dependent changes have been merged

## Further comments

If this is a relatively large or complex change, kick off the discussion at user@kylin or dev@kylin by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: examples/test_case_data/sandbox/kylin.properties [hunk 1]
```diff
@@ -183,7 +183,8 @@ kylin.web.default-time-filter=0
 # kylin query metrics percentiles intervals default=60, 300, 3600
 kylin.server.query-metrics-percentiles-intervals=60, 360, 3600
 
-# Env DEV|QA|PROD
+# Env DEV|QA|PROD\LOCAL\UT
```
Comment (bigxiaochu) [inline examples/test_case_data/sandbox/kylin.properties:186]: split by | ？

File: examples/test_case_data/sandbox/kylin.properties [hunk 2]
```diff
@@ -226,4 +227,4 @@ kylin.engine.flink-conf.yarn.nodelabel=
 ### QUERY PUSH DOWN  ###
 #kylin.query.pushdown.runner-class-name=org.apache.kylin.query.pushdown.PushDownRunnerSparkImpl
 
-kylin.query.transformers=org.apache.kylin.query.util.DefaultQueryTransformer
+kylin.query.transformers=org.apache.kylin.query.util.DefaultQueryTransformer
```
Comment (bigxiaochu) [inline examples/test_case_data/sandbox/kylin.properties:230]: I dont know what is modified ...

File: kylin-spark-project/kylin-spark-engine/src/main/java/org/apache/kylin/engine/spark/job/NSparkExecutable.java [hunk 3]
```diff
@@ -146,7 +146,7 @@ protected ExecuteResult doWork(ExecutableContext context) throws ExecuteExceptio
             throw new ExecuteException("meta dump failed", e);
         }
         String filePath = dumpArgs();
-        if (config.isUTEnv() || config.isZKLocal()) {
+        if (config.isUTEnv() || config.isLOCALEnv() || config.isZKLocal()) {
```
Comment (bigxiaochu) [inline kylin-spark-project/kylin-spark-engine/src/main/java/org/apache/kylin/engine/spark/job/NSparkExecutable.java:149]: UT is unittest, but local is local. So isLocalEnv()?

File: kylin-spark-project/kylin-spark-engine/src/main/java/org/apache/kylin/engine/spark/source/CsvSource.java [hunk 4]
```diff
@@ -59,7 +59,12 @@ public CsvSource(KylinConfig config) {
                         && (parameters != null && parameters.get("separator") == null)) {
                     path = "file:///" + new File(getUtMetaDir(),
                             "../../examples/test_case_data/parquet_test/data/" + table.identity() + ".csv")
-                                    .getAbsolutePath();
+                            .getAbsolutePath();
+                    separator = "";
+                } else if (kylinConfig.getDeployEnv().equals("LOCAL")) {
+                    path = "file:///" + new File(getUtMetaDir(),
+                            "../parquet_test/data/" + table.identity() + ".csv")
```
Comment (bigxiaochu) [inline kylin-spark-project/kylin-spark-engine/src/main/java/org/apache/kylin/engine/spark/source/CsvSource.java:66]: why ut path is "../../examples/test_case_data/parquet_test/data/", but local path is  "../parquet_test/data/"?
can they use same path?

File: examples/test_case_data/sandbox/kylin.properties (hunk 1 above)
Comment (RupengWang) [reply examples/test_case_data/sandbox/kylin.properties]: Sure, it's not so obvious in my IDE between '|' and '\' .

File: kylin-spark-project/kylin-spark-engine/src/main/java/org/apache/kylin/engine/spark/job/NSparkExecutable.java (hunk 3 above)
Comment (hit-lacus) [reply kylin-spark-project/kylin-spark-engine/src/main/java/org/apache/kylin/engine/spark/job/NSparkExecutable.java:149]: Agreed. Please capitalize **local** .

File: kylin-spark-project/kylin-spark-engine/src/main/java/org/apache/kylin/engine/spark/source/CsvSource.java (hunk 4 above)
Comment (hit-lacus) [reply kylin-spark-project/kylin-spark-engine/src/main/java/org/apache/kylin/engine/spark/source/CsvSource.java:66]: Agreed, these two path confusing me too.

**Comment by hit-lacus** [comment]
LGTM

**Pull request merged**